gradle-conjure/src/main/java/com/palantir/gradle/conjure/CompileIrTask.java

```diff
@@ -37,24 +38,40 @@
 @CacheableTask
 public class CompileIrTask extends DefaultTask {
     private static final String EXECUTABLE = OsUtils.appendDotBatIfWindows("bin/conjure");
+    private final RegularFileProperty outputIrFile = getProject().getObjects().fileProperty();
 
-    private File outputFile;
     private Supplier<File> inputDirectory;
     private Supplier<File> executableDir;
     private final SetProperty<ServiceDependency> productDependencies =
             getProject().getObjects().setProperty(ServiceDependency.class);
 
+    /**
+     * Eagerly set where to output the generated IR.
+     *
+     * @deprecated Use {@link #getOutputIrFile()} instead.
+     */
+    @Deprecated
     public final void setOutputFile(File outputFile) {
-        this.outputFile = outputFile;
+        outputIrFile.set(outputFile);
     }
 
-    public final void setInputDirectory(Supplier<File> inputDirectory) {
-        this.inputDirectory = inputDirectory;
+    /**
+     * Eagerly get where to output the generated IR.
+     *
+     * @deprecated Use {@link #getOutputIrFile()} instead.
+     */
+    @Deprecated
+    public final File getOutputFile() {
```

Comment:
think you want `@Internal` too or else gradle 6 complains

Reply:
I think it only yells at methods with Property return types

Reply:
oh i thought it's any method that starts with `get` i.e. that looks like a bean accessor